### PR TITLE
v6.0.0 Release:

### PR DESCRIPTION
**Pull request by Gauthamastro**
## Describe your changes
1. Introduction of Trading Fees
2. Liquidity Mining Program
3. Crowd-sourced Liquidity Mining
4. Orderbook v2 support
5. Bug fixes for orderbook APIs

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

## Summary by CodeRabbit

- **New Features**
	- Introduced liquidity mining functionality for managing liquidity pools, including adding and removing liquidity, and claiming rewards.
	- Added new RPC methods for querying account scores, rewards, trading volumes, and more in the OCEX pallet.
	- Implemented new functionality for managing trading pairs, market metrics, and rewards calculations.
	- Added integration tests for on-chain validation of trades and complete flow tests for specific market scenarios.
	- Introduced structures and configurations related to Liquidity Mining Program (LMP) and fee distribution.
- **Enhancements**
	- Updated testing configurations to include new pallets and functionalities.
	- Enhanced error handling for offchain HTTP requests.
	- Improved logic for fee calculations and updates in trading operations.
- **Bug Fixes**
	- Fixed visibility issues in the `FeeConfig` struct.
- **Documentation**
	- Updated licensing information and comments across various files for clarity.
- **Refactor**
	- Renamed types and modules for consistency and clarity.
	- Reorganized runtime configurations and dependencies.
- **Tests**
	- Added new tests for liquidity mining and trading functionalities.
- **Chores**
	- Updated spec versions for mainnet and parachain runtimes.
	- Adjusted minimum execution times and associated weights in weight files for various pallets.

### CODE REVIEW

File: pallets/ocex/src/validator.rs [hunk 1]
```diff
@@ -359,18 +712,132 @@
...
+		let mut config = get_lmp_config(state)?;
+		let next_index = config.index.saturating_add(1);
+		for (main, score) in scores {
+			store_q_score_and_uptime(
+				state,
+				config.epoch,
+				next_index,
+				*score,
+				&market,
+				&Decode::decode(&mut &main.encode()[..]).unwrap(), // unwrap is fine.
```
Comment (Gauthamastro) [inline pallets/ocex/src/validator.rs:757]: Hack for subxt types, won't fail

File: primitives/thea/src/types.rs [hunk 2]
```diff
@@ -18,8 +18,6 @@
 
 //! Definition of types used for `Thea` related operations.
 
-#![cfg_attr(not(feature = "std"), no_std)]
-
 use frame_support::__private::log;
 use parity_scale_codec::{Decode, Encode};
 use polkadex_primitives::UNIT_BALANCE;
```
[the remark this answers is not in the document]
Comment (Gauthamastro) [reply primitives/thea/src/types.rs:23]: It's fine, it should be defined in root level

File: pallets/ocex/src/lib.rs [hunk 3]
```diff
@@ -1505,6 +1252,855 @@
...
+				.map_err(|_| Error::<T>::TradingFeesBurnFailed)?;
+				// Emit an event
+				Self::deposit_event(Event::<T>::AuctionClosed {
+					bidder,
+					burned: Compact::from(amount_to_be_burnt),
+					paid_to_operator: Compact::from(trasnferable_amount),
+				})
+			}
+			Ok(())
+		}
```
[the remark this answers is not in the document]
Comment (Gauthamastro) [reply pallets/ocex/src/lib.rs:2102]: It's not a result

File: primitives/orderbook/src/lmp.rs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+/// All metrics used for calculating the LMP score of a main account
+#[derive(Decode, Encode, TypeInfo, Copy, Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TraderMetric {
+	pub maker_volume: Decimal, // Trading volume generated where main acc is a maker
+	pub fees_paid: Decimal,    // defined in terms of quote asset
+	pub q_score: Decimal,      // Market making performance score
+	pub uptime: u16,           // Uptime of market maker
+}
```
[the remark this answers is not in the document]
Comment (Gauthamastro) [reply primitives/orderbook/src/lmp.rs]: Not used, removed.

**Review by serhii-temchenko — APPROVED** [review]

**Review by nuel77 — APPROVED** [review]

**Review by zktony — APPROVED** [review]

**Pull request merged**